src/log/lss.rs

```diff
@@ -18,62 +20,81 @@ impl LockFreeLog {
 
         LockFreeLog { iobufs: iobufs }
     }
-}
-
-impl Log for LockFreeLog {
-    fn reserve(&self, buf: Vec<u8>) -> Reservation {
-        self.iobufs.reserve(buf)
-    }
 
-    /// return the config in use for this log
-    fn config(&self) -> &Config {
-        self.iobufs.config()
-    }
-
-    fn write(&self, buf: Vec<u8>) -> LogID {
-        self.iobufs.reserve(buf).complete()
-    }
-
-    /// read a buffer from the disk
-    fn read(&self, id: LogID) -> io::Result<Result<Vec<u8>, usize>> {
+    fn read_with_raw_sz(&self, id: LogID) -> (io::Result<Result<Vec<u8>, usize>>, usize) {
```

Comment:
needed this internal function for returning the raw length as well as the possible item, for use with the iterator. before, the iterator was advancing the next offset by the length of the message it read from the log, but when using compression this messes things up, and we need to advance a (usually much lower) number depending on the number of compressed bytes on disk